frontend/ui/lib/widgets/create_trip_widget.dart

```diff
@@ -1,27 +1,51 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:tripmeout/widgets/map_widget.dart';
+import 'package:tripmeout/services/trip_service.dart';
+import 'package:tripmeout/model/trip.dart';
+import 'package:tripmeout/model/location.dart';
+import 'package:tripmeout/router/router.dart';
 
 //TODO: Add loading screen after to the Create Trip Widget
 //TODO: Get rid of the newInformation text on the page.
 
 class CreateTripWidget extends StatefulWidget {
+  final TripService tripService;
+
+  CreateTripWidget(this.tripService);
+
   @override
   _CreateTripWidgetState createState() => _CreateTripWidgetState();
 }
 
 class _CreateTripWidgetState extends State<CreateTripWidget> {
-  String place = "No Input";
+  TripService get tripService => widget.tripService;
+
+  bool _enabled = false;
+
+  String place;
   int radius = 0;
   String newInformation = 'Grabbed info placed here.';
-  void changeText() {
+  void submitTrip() {
     setState(() {
+      tripService.createTrip(Trip(
```

Comment:
Be careful: this returns a future. Since we aren't awaiting its result, we're ignoring failures that might happen on submission and assuming that it was successful.

We should definitely `await` the result. I'm not sure what the implications are of running an `async` function in `setState()`, but it's probably safer to do this _before_ the `setState` call (and await the results), and then just set `newInformation` within `setState`.